### ISSUE TITLE
ec2 service discovery works incorrectly

### ISSUE THREAD

**Issue by f41gh7** [VictoriaMetrics/VictoriaMetrics#771]
**Describe the bug**

 vmagent doesn't support all prometheus ec2_sd_config features.

 With prometheus ec2_sd_config is possible to use following auth methods in addition to AWS_KEY and AWS_SECRET vars:
- role_arn - sets explicit at ec2_sd_config
- instance iam role - sets implicit, when iam role assigned to ec2 instance and AWS_KEY with AWS_SECRET configuration isn't set.

**To Reproduce**
 based on this [article](https://jarodw.com/posts/prometheus-ec2-sd-multiple-aws-accounts/)
 Assign iam role to instance and create following configuration:

```yaml
scrape_configs:
  - job_name: 'node'
    ec2_sd_configs:
      - region: eu-west-1
    # Discover instances in account 111111111111
    - port: 9100
      role_arn: arn:aws:iam::111111111111:role/prometheus-assume-role-PrometheusAssumeRole-KJ4TK9KJBPU7 

```
 with prometheus it works, vmagent returns error: 
```missing `access_key` in AWS_ACCESS_KEY_ID env var; probably, `access_key` must be set in `ec2_sd_config`?```

**Expected behavior**

 ec2_sd_config must return instances for scraping.

**Version**
1.41.0

**Comment by valyala** [comment]
`vmagent` should be able to work properly with `role_arn` starting from the commit 312fead9a2e261d470858b8d1ffec7527682dc77 . This commit will be included in the next release.

**Comment by valyala** [comment]
FYI, support for `role_arn` and EC2 instance tokens has been included in [v1.41.1](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing this bug as fixed.
